### PR TITLE
Adding 3 new step templates: Remove server from Azure Load Balancer, Wait for IIS Connections to drop to 0 or time limit, and Put server back in Azure Load Balancer.

### PR DESCRIPTION
**Pull request by tbrasch**

### CODE REVIEW

File: step-templates/put-server-back-in-azure-load-balancer.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
+{
+  "Id": "540de6f6-f1c6-4e1c-9062-9e78570074f6",
+  "Name": "Put Server Back In Azure Load Balancer",
+  "Description": "Put Server Back In Azure Load Balancer",
```
Comment (alfhenrik) [inline step-templates/put-server-back-in-azure-load-balancer.json]: As this step template requires the use of a Service Principal, could you mention that in the description?

File: step-templates/remove-server-from-azure-load-balancer.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
+{
+  "Id": "5658d525-2a04-47da-85a0-00244976d811",
+  "Name": "Remove Server From Azure Load Balancer",
+  "Description": "Remove Server From Azure Load Balancer",
```
Comment (alfhenrik) [inline step-templates/remove-server-from-azure-load-balancer.json]: Same here, as this step template requires the use of a Service Principal, could you mention that in the description?

File: step-templates/put-server-back-in-azure-load-balancer.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
... be null.'\n}\n$azureLoadBalancer = $OctopusParameters['psbilbAzureLoadBalancer']\nWrite-Host ('Azure Load Balancer: ' + $azureLoadBalancer)\n\n#endregion\n\n\n#region Process\n\nTry\n{\n  Get-AzureRmContext | Out-Null\n}\nCatch\n{\n  If ($_.Exception.ToString() -like \"*Run Login-AzureRmAccount to login.*\")\n  {\n    Login-AzureRmAccount\n  }\n}\n\nSet-AzureRmContext -SubscriptionName $azureSubscription\n\n$azureVM = Get-AzureRmVM -ResourceGroupName $azureResourceGroup -Name $azureMachineName\nIf (!$azureVM)\n{\n  Throw 'Could not retrieve server from Azure needed to remove from Load Balancer.'\n}\n\n$nic = (Get-AzureRmNetworkInterface -ResourceGroupName $azureResourceGroup | Where-Object {$_.VirtualMachine.Id -eq $azureVM.Id})\nIf (!$nic)\n{\n  Throw 'Could not retrieve NIC from Azure needed to remove from Load Balancer.'\n}\n\n$loadBalancer = Get-AzureRmLoadBalancer -Name $azureLoadBalancer -ResourceGroupName $azureResourceGroup\nIf (!$loadBalancer)\n{\n  Throw 'Could not retrieve Load Balancer info from Azure.'\n}\n\n$nic.IpConfigurations[0].LoadBalancerBackendAddressPools = $loadBalancer.BackendAddressPools\nSet-AzureRmNetworkInterface -NetworkInterface $nic\n\n#endregion\n",
```
Comment (alfhenrik) [inline step-templates/put-server-back-in-azure-load-balancer.json]:
```
Try
{
  Get-AzureRmContext | Out-Null
}
Catch
{
  If ($_.Exception.ToString() -like "*Run Login-AzureRmAccount to login.*")
  {
    Login-AzureRmAccount
  }
}
```
I don't think this should happen as our bootstrap script handles all the authentication to Azure, **but** could there be a chance here that the `Login-AzureRmAccount` would open a dialog and cause the deployment to hang?

**Comment by tbrasch** [comment]
@alfhenrik - ok, I think it should be good now. Please let me know if it looks good to you. Thanks!

**Pull request merged**